pkg/authorization/authorizer/scope/converter.go

```diff
@@ -130,43 +140,54 @@ func (clusterRoleEvaluator) Handles(scope string) bool {
 }
 
 func (e clusterRoleEvaluator) Validate(scope string) error {
-	_, _, err := e.getRoleNamespace(scope)
+	_, _, _, err := e.parseScope(scope)
 	return err
 }
 
-func (e clusterRoleEvaluator) getRoleNamespace(scope string) (string, string, error) {
+func (e clusterRoleEvaluator) parseScope(scope string) (string /*role name*/, string /*namespace*/, bool /*escalating*/, error) {
 	if !e.Handles(scope) {
-		return "", "", fmt.Errorf("bad format for scope %v", scope)
+		return "", "", false, fmt.Errorf("bad format for scope %v", scope)
+	}
+	escalating := false
+	if strings.HasSuffix(scope, ":!") {
+		escalating = true
+		// clip that last segment before parsing the rest
+		scope = scope[:strings.LastIndex(scope, ":")]
```

Comment:
> As it stands, doing the check on the length of the token array after doing a SplitN seems silly ...

You can't count the number of colons because role names are allowed to contain colons.